tests/manage/z_cluster/cluster_expansion/test_add_node_crashcollector.py

```diff
@@ -0,0 +1,136 @@
+    8.Wait for 3 osd pods to be on running state
+    9.Verify ceph-crashcollector pod running on worker node where "rook-ceph" pods are running.
+
+    """
+
+    def test_add_node_crash_collector(self, add_nodes, node_drain_teardown):
+        """
+        Add node with OCS label and verify crashcollector created on new node
+
+        """
+        logger.info("Get Node name where mgr pod running")
+        mgr_pod_nodes = [get_pod_node(pod) for pod in get_mgr_pods()]
+        mgr_pod_node_name = [node.name for node in mgr_pod_nodes]
+
+        logger.info("Add one worker node with OCS label")
+        add_nodes(ocs_nodes=False, node_count=1)
+        ceph_cluster_obj = CephCluster()
+        assert ceph_cluster_obj.wait_for_rebalance(
+            timeout=3600
+        ), "Data re-balance failed to complete"
+
+        drain_nodes(mgr_pod_node_name)
+
+        logging.info("Wait for 3 mon pods to be on running state")
+        pod = OCP(kind=constants.POD, namespace=config.ENV_DATA["cluster_namespace"])
+        assert pod.wait_for_resource(
+            condition="Running",
+            selector=constants.MON_APP_LABEL,
+            resource_count=3,
+            timeout=1400,
+        )
+        logger.info(
+            "Verify rook-ceph-crashcollector pod running on worker node"
+            " where rook-ceph pods are running."
+        )
+        assert sorted(self.get_crashcollector_nodes()) == sorted(
+            self.get_nodes_where_ocs_pods()
+        ), (
+            f"The crashcollector pod does not exist on "
+            f"{self.get_nodes_where_ocs_pods() - self.get_crashcollector_nodes()} "
+            f"even though rook-ceph pods are running on this node"
+        )
+
+        schedule_nodes(mgr_pod_node_name)
+
+        logging.info("Wait for 3 osd pods to be on running state")
+        assert pod.wait_for_resource(
+            condition="Running",
+            selector=constants.OSD_APP_LABEL,
+            resource_count=3,
+            timeout=600,
+        )
+
+        logger.info(
+            "Verify rook-ceph-crashcollector pod running on worker node where rook-ceph pods are running."
+        )
+        assert sorted(self.get_crashcollector_nodes()) == sorted(
+            self.get_nodes_where_ocs_pods()
+        ), (
+            f"The crashcollector pod does not exist on "
+            f"{self.get_nodes_where_ocs_pods() - self.get_crashcollector_nodes()} "
+            f"even though rook-ceph pods are running on this node"
+        )
+
+    def get_crashcollector_nodes(self):
+        """
+        Get the nodes names where crashcollector pods are running
+
+        return:
+            set: cluster names where crashcollector pods are running
+
+        """
+        crashcollector_pod_names = get_pod_name_by_pattern(pattern="crashcollector")
+        crashcollector_pod_objs = [
+            get_pod_obj(crashcollector_pod_name)
+            for crashcollector_pod_name in crashcollector_pod_names
+        ]
```

Comment:
instead of getting the pods by its name pattern , IMO it is better to get the crash collector pods with its label "app=rook-ceph-crashcollector"
You can refer to similar implementation how it is done for other ocs pods, wdyt?
https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/resources/pod.py#L940

Reply:
yes. I agree. 
```
$ oc -n openshift-storage get Pod  -n openshift-storage --selector=app=rook-ceph-crashcollector
NAME                                                  READY   STATUS    RESTARTS   AGE
rook-ceph-crashcollector-compute-0-5dbfcf589-lp4nz    1/1     Running   0          3d2h
rook-ceph-crashcollector-compute-1-6f788dcc98-7t5k2   1/1     Running   0          2d5h
rook-ceph-crashcollector-compute-2-56fbbd9587-stmfr   1/1     Running   0          3d2h

```

Reply:
done